pkg/k8sutil/deployments.go

```diff
@@ -160,14 +160,19 @@ func (k *K8sutil) CreateClientDeployment(baseImage string, replicas *int32, java
 						},
 					},
 					Spec: v1.PodSpec{
+						SecurityContext: &v1.PodSecurityContext{
+							RunAsUser: &k.RunAsUser,
```

Comment:
What if these aren't defined how does the controller react? We should add tests around them to validate this new behavior.

Reply:
The default for RunAsUser (0) is defined in init(), so there's always a value.